pkg/game/board.go

```diff
@@ -139,14 +109,35 @@ func (board *board) GetLegalMovesFor(tile elements.PlacedTile) []elements.Placed
 	return []elements.PlacedTile{}
 }
 
+// Returns true if the tile placement position is valid, i.e. if all existing neighbouring tiles have matching features.
+// (for example, city feature directly neighbouring road or field is not valid)
+// Does not take meeples into account.
 func (board *board) isPositionValid(tile elements.PlacedTile) bool {
-	for _, f := range tile.Features {
-		s := side.Top
-		for range 4 {
-			if f.Sides.HasSide(s) && !board.testSide(tile.Position, s.Rotate(2), f.FeatureType) {
+	for _, tileFeature := range tile.Features {
+		for _, side := range side.EdgeSides {
+			if tileFeature.Sides.HasSide(side) {
+				neighbourPosition := position.FromSide(side).Add(tile.Position)
+				neighbouringTile, exists := board.GetTileAt(neighbourPosition)
+				if exists && neighbouringTile.GetPlacedFeatureAtSide(side.Mirror(), tileFeature.FeatureType) == nil {
+					return false
+				}
+			}
+
+		}
+	}
+	// roads are checked separately because when placing a tile with fields neighbouring a tile with roads,
+	//  features checked from this tile to the neighbour are still matching (every field has a corresponding field on the neighbour tile),
+	//  but roads from the other tile might still not have a corresponding road in this one, despite the previous check returning true
+	//
+	// TODO: figure out a cleaner way to do this
```

Comment:
To be honest, once I understood why this was needed, I no longer considered it unclean. The original comment focuses on how this is technically specific to roads but that's really just an optimization strategy - the general idea is that due to features potentially having overlaps, it is not enough to check that our features have counterparts since the neighbour could also have **another** feature on the side that we checked.

On a high-level, we're doing:
```py
# input: tile
for side in EDGE_SIDES:
    neighbour = board.neighbour(tile, side)
    if neighbour and tile.features_on(side) != neighbour.features_on(side.mirror()):
        return False

return True
```
It's just that our current APIs don't have a way to express this without affecting performance negatively.